bootstrap/012_task_log.sql

```diff
@@ -0,0 +1,54 @@
+
+create or replace function internal.task_run_id()
+returns text
+AS
+$$
+     SYSTEM$TASK_RUNTIME_INFO('CURRENT_TASK_GRAPH_RUN_GROUP_ID')
+$$;
+
+let task_log_exists boolean := (exists (select * from information_schema.tables where table_schema = 'INTERNAL' and table_name = 'TASK_LOG'));
+
+-- Generic table that tasks should record their execution into.
+CREATE TABLE INTERNAL.TASK_LOG IF NOT EXISTS (task_start timestamp_ltz, success boolean, task_name varchar, object_name varchar,
+    input variant, output variant, task_finish timestamp_ltz, task_run_id text, query_id text);
+
+create or replace procedure internal.migrate_task_logs()
+    returns text
+    language sql
+AS
+BEGIN
+    -- If the old task log tables exist, copy the rows into the new task_log table
+    if (exists (select * from information_schema.tables where table_schema = 'INTERNAL' and table_name = 'TASK_QUERY_HISTORY')) then
```

Comment:
It looks like we will copy data from the old to the new if the old exists. Where do we delete the old and/or prevent multiple runnings of this creating dub data

Reply:
Oh, duh. I was thinking about the "next" upgrade and didn't think about the "next-next" upgrade.

Reply:
will truncate old log tables after insert.